elife/multiservice.sls

```diff
@@ -0,0 +1,45 @@
+
+# systemd managed group processes
+
+# 1. N number of the given process are running
+# 2. the pool of processes can be grown or shrunk
+# 3. broken processes fail highstate as they ordinarily would
+
+{% for process, opts in pillar.elife.multiservice.services.items() %}
+    {% set num_processes = opts["num_processes"] %}
+
+# ensure the target state is available and running
+{{ process }}-controller.target:
+    file.managed:
+        - name: /lib/systemd/system/{{ process }}-controller.target
+        - source: salt://elife/templates/process-controller.target
+
+# ensure precisely N processes are enabled. restart controller if N has changed
+# enabled processes are started when the controller state changes
+{{ process }}-set-enabled:
+    cmd.run:
+        - name: |
+            set -e
+            systemctl stop {{ process }}@{0..99} # stop any running instances of this service.
+            systemctl disable {{ process }}@ # disable *everything*. implicit reload
```

Comment:
service definition file isn't reloaded if the number of processes doesn't change, while it should happen for any change to that file

Reply:
Solved by separating the `enable` part so that it is always executed (it's idempotent).

Reply:
I'll take a look at your work today